src/Disks/S3/DiskS3.cpp

```diff
@@ -675,11 +675,12 @@ void DiskS3::readRestoreInformation(DiskS3::RestoreInformation & restore_informa
 {
     const ReadSettings read_settings;
     auto buffer = metadata_disk->readFile(RESTORE_FILE_NAME, read_settings, 512);
-    buffer->next();
-
-    try
+    if (buffer)
```

Comment:
Wrong modification.

Reply:
`readFile` will never return nullptr, either a valid value or exception is thrown.